### PR TITLE
planner: fix do-statement will error with agg function

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31413

### What is changed and how it works?

Reason: In normal build-phase, the aggFunc is built ahead of other select fields, informing the latter one to skip rewrite progress of the agg field via aggMapper. While we miss this when handling do-statement.

Like comments down on the issue, do-statement can only contain expressions, in which the aggFunc may take place, leading the name-resolve problem.

Approach: build the aggMapper first like what it does in the building-select

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -907,6 +938,16 @@ func (b *PlanBuilder) buildCreateBindPlan(v *ast.CreateBindingStmt) (Plan, error
 	return p, nil
 }
 
+// detectDoAgg detects an aggregate function in its exprs.
```
Comment (xuyifangreeneyes) [inline planner/core/planbuilder.go:941]: The function name in comment is different.

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -787,8 +787,39 @@ func (b *PlanBuilder) buildDo(ctx context.Context, v *ast.DoStmt) (Plan, error)
 	proj := LogicalProjection{Exprs: make([]expression.Expression, 0, len(v.Exprs))}.Init(b.ctx, b.getSelectOffset())
 	proj.names = make([]*types.FieldName, len(v.Exprs))
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(v.Exprs))...)
+
+	// Since do statement only contain expression list, and it may contain aggFunc, detect the build the aggMapper firstly.
```
Comment (xuyifangreeneyes) [inline planner/core/planbuilder.go:791]: The last sentence seems have grammar error.

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**